[docs] misc updates for 1.5 release (#964)
agentgateway/website: content/docs/standalone/main/configuration/security/external-authz.md

## Conditional execution

To choose between multiple external authorization servers based on the request, use the `conditional` field. For example, you can send admin paths to a stricter authorization server and route every other request to a standard one. For details, see [Conditional policies]({{< link-hextra path="/configuration/policies/conditional-policies" >}}).

## Connection-level external authorization {#network-extauthz}

The `extAuthz` policy calls the authorization service once for each HTTP request. To call it once for each downstream connection instead, use the `networkExtAuthz` frontend policy.

> Use connection-level authorization in two cases.

Reviewer: ```suggestion
Scoping external authorization to a downstream connection is useful in the following cases.
```

- The gateway carries TCP traffic that has no HTTP requests to authorize.
- The connection is long-lived, and a callout on every request costs more than the decision is worth.

> Configure it under `frontendPolicies.networkExtAuthz`. It takes the same fields as `extAuthz`, with one restriction.

Reviewer: ```suggestion
Configure connection-level authorization under the `frontendPolicies.networkExtAuthz` section. The section takes the same fields as `extAuthz`, with one restriction.
```

> [!IMPORTANT]
> The `networkExtAuthz` policy calls the authorization service over HTTP only, so you must set `protocol.http` explicitly. The `protocol` field defaults to `grpc`, which means that a configuration that omits it fails to start with `frontendPolicies.networkExtAuthz only supports protocol.http`.

```yaml
# yaml-language-server: $schema=https://agentgateway.dev/schema/config
frontendPolicies:
  networkExtAuthz:
    host: localhost:9000
    protocol:
      http: {}

gateways:
  default:
    port: 3000
routes:
- backends:
  - host: localhost:8080
```

> Because the policy runs before protocol handling, it also applies to a gateway that carries no HTTP traffic.

Reviewer: ```suggestion
Because the policy runs before protocol handling, it also applies to a gateway that carries non-HTTP traffic.
```

```yaml
# yaml-language-server: $schema=https://agentgateway.dev/schema/config
frontendPolicies:
  networkExtAuthz:
    host: localhost:9000
    protocol:
      http: {}

gateways:
  default:
    port: 3000
    protocol: TCP
tcpRoutes:
- gateways: [default]
  backends:
  - host: localhost:8080
```

The two policies differ in what the authorization service sees and in what a denial affects.

> | | `extAuthz` | `networkExtAuthz` |

Reviewer: this comparison kind of comes out of nowhere as this topic is really about `networkExtAuthz` and not `extAuthz`. maybe this should go into a separate heading with some more explanation about what you are comparing here
